Upi_Info_Card: Created a Upi Info card component

NileshSingh9245 opened the pull request:
### Release

1.0.0
https://blackminystudio.atlassian.net/browse/TPA-47

### Description

- Created a new component Upi info card,

### Screenshots

![image](https://github.com/user-attachments/assets/fb13dbeb-f3e7-411f-bc54-24dc6501d1df)

### Tips

Use this code in main.dart for testing.
```Dart
class HomeScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) => const Scaffold(
  backgroundColor: theme.colors.light,
        body: Center(
          child: Padding(
            padding: EdgeInsets.all(4.0),
            child: UpiInfoCard(), 
          ),
        ),
      );
}

```

---

### IMPORTANT:

- (REQUIRED) Coverage Screenshots for the tests written
- (REQUIRED) UI Screenshots for the UI implementation written

### Author preparation checklist

- [x] Current branch is rebased with latest 'main' branch
- [x] PR is targeting only one issue
- [x] Commit messages are descriptive, consistent, and follow the project's guidelines.
- [ ] Branch name matches with at least `<jira link reference>-<short change description>`
- [x] Primary reviewer is set as assignee to the PR
- [x] No unnecessary changes on `pubspec.lock`
- [x] All added TODOs have associated Github issue references
- [x] \*.iml and, pubsepc_overrides.yaml are not checkedin

### Primary reviewer checklist

- [x] All items in the author checklist is checked or not applicable
- [x] All files including .lock and assets are reviewed
- [x] Reasonable UI flows/cases related to the change are manually tested
- [x] No commented-out code or unnecessary debug statements.
- [x] No spelling or grammatical errors in code comments or documentation.
- [x] No sensitive data (e.g., API keys, credentials) is hardcoded or exposed.
- [x] The branch is up-to-date with the target branch (e.g., staging).

### Pre-merge checklist

- [ ] Two approvals received. (Changes affecting automatic PR verification, like new linter rules, formatting rules, Github workflow configurations)

---

NileshSingh9245 committed:
Upi_Info_Card: Created a Upi Info card component

quiteachaos commented on lib/widgets/upi_info_card.dart:
```diff
@@ -0,0 +1,108 @@
+                      keyboardType: TextInputType.number,
+                      onChanged: (value) => setState(() {}),
+                      decoration: InputDecoration(
+                        hintText: 'Enter Amount',
```
All the hard-coded texts need to be written in string_constants.dart inside widgets folder

NileshRajput1 replied:
Done

quiteachaos commented on lib/widgets/upi_info_card.dart:
```diff
@@ -0,0 +1,108 @@
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  Text(
+                    'Message:',
```
write in string_constants.dart and use it

NileshRajput1 replied:
Done

quiteachaos commented on lib/widgets/upi_info_card.dart:
```diff
@@ -0,0 +1,108 @@
+                      controller: _messageController,
+                      onChanged: (value) => setState(() {}),
+                      decoration: InputDecoration(
+                        hintText: 'Enter Message',
```
write in string_constants.dart and use it

NileshRajput1 replied:
Done

quiteachaos commented on lib/widgets/upi_info_card.dart:
```diff
@@ -0,0 +1,108 @@
+                  ),
+                ],
+              ),
+              SizedBox(height: theme.sizing.width.s2),
```
theme.spacing.width.s12 as per figma

quiteachaos commented on lib/widgets/upi_info_card.dart:
```diff
@@ -0,0 +1,108 @@
+              ),
+              SizedBox(height: theme.sizing.width.s2),
+              const Divider(),
+              SizedBox(height: theme.sizing.width.s2),
```
theme.spacing.width.s12 as per figma

NileshRajput1 replied:
done

NileshSingh9245 committed:
UpiCardInfo and String Constants: Updated the required changes

NileshSingh9245 committed:
UpiInfoCard: Removed the Card Widget.

NileshRajput1 commented on lib/widgets/upi_info_card.dart:
```diff
@@ -0,0 +1,108 @@
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    return Card(
```
done

NileshSingh9245 commented on lib/widgets/upi_info_card.dart:
```diff
@@ -0,0 +1,108 @@
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(theme.borderradius.medium),
+      ),
+      child: Container(
```
Done

NileshSingh9245 committed:
Merge branch 'staging_1.0.0' into feature/upi_info_card

NileshSingh9245 committed:
UpiInfoCard: corrected border color and background colour of the component

NileshSingh9245 committed:
Merge branch 'feature/upi_info_card' of https://github.com/blackminystudio/tacklepay into feature/upi_info_card

NileshSingh9245 committed:
UpiInfoCard: Resolved Format

NileshSingh9245 committed:
Upi Info Card:
- Fixed card height and alignment to match Figma design.
- Ensured '₹' symbol is visible in the Amount field.
- Restricted Amount input to two decimal places and digits only using InputFormatters.
- Set consistent widths for both text fields; added ellipsis for overflow text.
- Implemented FocusNode to shift focus to the Message field after pressing 'Enter' in Amount field.
- Updated Divider thickness to 1px and color to contrastLow using a Container.

NileshSingh9245 commented:
**Use this code in main.dart for testing .**

```
class HomeScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Scaffold(
      backgroundColor: theme.colors.light,
      body: const Center(
          child: Padding(
        padding: EdgeInsets.symmetric(horizontal: 40),
        child: UpiInfoCard(),
      )),
    );
  }
}
```


**below is the list of addressed changes:**

> - Fixed card height and alignment to match Figma design.
> - Ensured '₹' symbol is visible in the Amount field.
> - Restricted Amount input to two decimal places and digits only using InputFormatters.
> - Set consistent widths for both text fields; added ellipsis for overflow text.
> - Implemented FocusNode to shift focus to the Message field after pressing 'Enter' in Amount field.
> - Updated Divider thickness to 1px and color to contrastLow using a Container.

satyabratanayak committed:
Rebuild Widget UPIinfoCard

satyabratanayak committed:
dart format

satyabratanayak committed:
flutter analyze --fatal-infos .

satyabratanayak committed:
format

quiteachaos commented on lib/widgets/upi_info_card.dart:
```diff
@@ -0,0 +1,252 @@
+  }
+}
+
+// TODO:
```
Remove the commented code

satyabratanayak replied:
done

quiteachaos commented on lib/widgets/string_constants.dart:
```diff
@@ -1,7 +1,19 @@
 const expenseText = 'Expense';
 const checkZero = '0';
 const checkMinus = '-';
+const newExpense = 'New Expense';
```
newExpenseText*

quiteachaos commented on lib/widgets/string_constants.dart:
```diff
@@ -1,7 +1,19 @@
+// TacklePay App
 const appName = 'Tackle Pay';
+const ruppeeSymbol = '₹';
```
rupeeSymbol

satyabratanayak replied:
done

quiteachaos commented on lib/widgets/upi_info_card.dart:
```diff
@@ -0,0 +1,252 @@
+    final length = number.length;
+    if (length <= 3) return number;
+
+    // Take the last three digits
```
redundant comment should be removed

satyabratanayak replied:
done

satyabratanayak committed:
remove unnecessary codes

satyabratanayak committed:
dart format .

NileshSingh9245 commented:
everything looks good to me

satyabratanayak committed:
Resolve QA Fail2

satyabratanayak committed:
Merge branch 'staging_1.0.0' into feature/upi_info_card
